lib/fauna/errors.rb

```diff
@@ -1,11 +1,33 @@
 module Fauna
   ##
+  # Error for when the server returns an unexpected kind of response.
+  class UnexpectedError < RuntimeError
+    # RequestResult for the request that caused this error.
+    attr_reader :request_result
+
+    def initialize(description, request_result)
+      # TODO: Should include more info here.
```

Comment:
I'm not sure what else we would want to put in here. I don't want to put the response in, nor really the status code.